weaknesses/MASVS-STORAGE/MASWE-0004.md

```diff
@@ -21,3 +21,25 @@ status: draft
 
 ---
 
+## Overview
+
+iOS and Android automatically back up app's to the cloud. These files often include sensitive user and app data. Under certain conditions, the backup may not be adequately secured by the cloud provider, or a malicious actor may tamper with the backed-up files. It's a good practice to review how your app instructs the system to perform backups and exclude sensitive files. This page also covers local backups made to your laptop and device-to-device transfers.
+
+## Impact
+
+- **Modification of app's behavior**: An attacker can tamper with data inside the backup, altering the app's logic. For example, they could modify a database that tracks the state of premium features and then restore the modified backup to the device. Another common scenario is backing up the device before redeeming a one-time coupon and restoring the backup afterward. This would allow the malicious actor to reuse the same coupon multiple times.
+
+- **Loss of confidentiality**: An attacker can extract sensitive data stored in the backup, such as personal information, photos, documents, and audio files.
+
+- **Loss of secure material**: An attacker can extract passwords, cryptographic keys, and session tokens to facilitate additional attacks, such as identity theft or account takeover.
+
+## Modes of Introduction
+
+- **System backs up app's data by default**: Backing up the device to the cloud is enabled by default for all apps once the user gives consent during setting up the phone for the first time.
+- **Users can back up app's data locally to a laptop**: Under certain conditions, an attacker can tamper the backup locally.
+
+- **Usage of Device-To-Device transfer**: Transferring data to another device enables an attacker to perform similar attacks.
```

Comment:
I'd suggest this to improve the content. However, I'm thinking that this should be more in the intro, and here we should focus more on how this can happen because the developer did or didn't do something. Typically, "Modes of Introduction" represents those things and helps us define the tests. Maybe it's more like

- Not preventing automatic system backups
- Sensitive data not excluded from backups
- Unencrypted backups (if the app creates custom backups)

What do you think?

```suggestion
- **Automatic System Backups**: By default, iOS and Android back up app data to the cloud once the user consents during the initial setup.

- **Local Backups**: Users can back up their devices to local systems (e.g., laptops). If local backups are stored unencrypted or not securely handled, attackers could tamper with this data.

- **Device-To-Device Transfer**: Transferring data between devices (e.g., via iCloud or Google’s device-to-device migration tools) enables an attacker to perform similar attacks.
```